packages/adminpanel-v2/src/stores/CustomerStore.ts

```diff
@@ -0,0 +1,84 @@
+//import { TCustomer, TProject, TActivity} from "$lib/types"
+import type { TCustomer } from "$lib/types";
+import { get, writable } from "svelte/store";
+import { customers as mock}  from "$lib/mock/customers";
+
+type TCustomerStore = {
+	customers: TCustomer[],
+	active: {
+		customer: number | undefined,
+		project: number | undefined,
+		activity: number | undefined
+	}
+}
+
+function createCustomers() {
+	const customerStore = writable<TCustomerStore>(
+		{
+			customers: mock,
+			active: {
+				customer: undefined,
+				project: undefined,
+				activity: undefined
+			}
+		}
+	);
+	const { subscribe, set, update } = customerStore;
+
+	const fetchCustomers = () => {
+		//fetch from backend
+	}
+
+	const setCustomers = () => {
+		update((n) => {n.customers = mock; return n})
+		
+	}
+
+    const setCustomer = (id: number) => {
+		if (id != get(customerStore).active.customer) {
+			update((n) => {n.active.customer = id; return n})
+			update((n) => {n.active.project = undefined; return n})
+			update((n) => {n.active.activity = undefined; return n})
+		}
+	}
+
+	const setProject = (id: number) => {
```

Comment:
Same as above, when we setProject, do we write to a temporary memory per now? And is the store a kind of temporary memory during dev? Or do you have to write to a database directly?

And when you write to store, is there any hard constraints on how much you can write?

Reply:
The store is responsible for maintaining the state of the frontend. Once connected to the backend, async functions must be added to update the data in the backend, wait for a response, and then update the frontend store. Such functions will be added directly in the store, so backend connection is not available without going through the store.

Reply:
Thanks for a good a extensive explanation!